.github/workflows/release.yaml

```diff
@@ -1,43 +1,85 @@
+    name: Prepare release
     runs-on: ubuntu-latest
     permissions:
-      contents: read
+      actions: write
+      contents: write
       packages: write
 
     steps:
-    - name: get latest release with tag
-      id: latestrelease
-      run: |
-          echo "::set-output name=releasetag::$(curl -s https://api.github.com/repos/OCTRI/authentication-lib/releases/latest | jq '.tag_name' | sed 's/\"//g')"
+    - name: Check out repository
+      uses: actions/checkout@v4
+
+    - name: Set up Java build environment
+      uses: actions/setup-java@v4
+      with:
+        distribution: 'temurin'
+        java-version: '17'
+        cache: 'maven'
 
-    - name: confirm release tag
+    - name: Set up Git
       run: |
-          echo ${{ steps.latestrelease.outputs.releasetag }}
+        git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git config --global user.name "github-actions[bot]"
 
-    - name: tagcheckout
-      uses: actions/checkout@v4
+    - name: Update CHANGELOG.md
+      id: update_changelog
+      uses: thomaseizinger/keep-a-changelog-new-release@3.1.0
       with:
-          ref: ${{ steps.latestrelease.outputs.releasetag }}
+        tag: v${{ inputs.version }}
 
-    - name: Set up JDK 17
-      uses: actions/setup-java@v4
+    - name: Commit CHANGELOG.md
+      run: |
+        git add CHANGELOG.md
+        git commit -m "Add version ${{ inputs.version }} to CHANGELOG.md"
+
+    - name: Prepare Maven release
+      run: |
+        mvn -B \
+          -Darguments=-DskipTests \
+          -DautoVersionSubmodules=true \
+          -DreleaseVersion=${{ inputs.version }} \
+          -DscmReleaseCommitComment="Release version ${{ inputs.version }}" \
+          -DdevelopmentVersion=${{ inputs.development_version }}-SNAPSHOT \
+          -DscmDevelopmentCommitComment="Next development version - ${{ inputs.development_version }}" \
+          -Dtag=v${{ inputs.version }} \
+          -Dusername=${{ github.actor }} \
+          -Dpassword=${{ secrets.GITHUB_TOKEN }} \
+          release:prepare
+
+    - name: Create GitHub release
+      uses: softprops/action-gh-release@v2
       with:
-        java-version: '17'
-        distribution: 'temurin'
-        server-id: github
-        gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
-        gpg-passphrase: MAVEN_GPG_PASSPHRASE
+        body: ${{ steps.update_changelog.outputs.release-notes }}
+        draft: false
+        prerelease: false
+        name: ${{ inputs.version }}
+        tag_name: v${{ inputs.version }}
+        generate_release_notes: false
 
-    - name: Publish to GitHub Packages Apache Maven
-      run: mvn deploy -DskipTests -Pci
+    - name: Build release artifacts
+      run: |
+        gh workflow run build.yaml --ref v${{ inputs.version }}
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build snapshot artifacts
+      run: |
+        gh workflow run build.yaml --ref main
```

Comment:
This reuses the existing build workflow to produce the binary artifacts, rather than repeating the logic here.